listComprehensions.py

```diff
@@ -22,8 +22,6 @@ def main():
     full_initials = [(name.split()[0][0], name.split()[1][0]) for name in full_names]
     print(full_initials)
 
-
```

Comment:
That's odd, there is no gap;

(Also attached a screen-print of my code)


Also it seems that something happened to the comment and def main is lower down then it should, any idea as to why that's the case?.


"""
CP1404/CP5632 Practical
List comprehensions
"""

def main():
    names = ["Bob", "Angel", "Jimi", "Alan", "Ada"]
    full_names = ["Bob Martin", "Angel Harlem", "Jimi Hendrix", "Alan Turing", "Ada Lovelace"]

    # for loop that creates a new list containing the first letter of each name
    first_initials = []
    for name in names:
        first_initials.append(name[0])
        print(first_initials)

    # list comprehension that does the same thing
    first_initials = [name[0] for name in names]
    print(first_initials)

    # list comprehension that creates a list containing tuples of both initials
    # splits each name and adds the first letter of each part to a tuple
    full_initials = [(name.split()[0][0], name.split()[1][0]) for name in full_names]
    print(full_initials)

    almost_numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']

    # List comprehension that creates a list of integers from the above strings
    numbers = [number for number in almost_numbers]
    print(numbers)

    # List comprehension that creates a list of the full_names above in lowercase
    lowercase_full_names = [full_name.lower() for full_name in full_names]
    print(lowercase_full_names)


if __name__ == "__main__":
        main()


________________________________
From: AdamBorg <notifications@github.com>
Sent: Saturday, August 27, 2016 10:04:22 AM
To: garethmullins/Python-Week4Practicals
Cc: Gareth Mullins; State change
Subject: Re: [garethmullins/Python-Week4Practicals] Code review request (#1)


In listComprehensions.py<https://github.com/garethmullins/Python-Week4Practicals/pull/1#discussion_r76504709>:

> @@ -22,8 +22,6 @@ def main():
>      full_initials = [(name.split()[0][0], name.split()[1][0]) for name in full_names]
>      print(full_initials)
>
> -


Why is this huge gap here? Try keep code neat for readability,

-
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub<https://github.com/garethmullins/Python-Week4Practicals/pull/1/files/09bb1e501bd70aedba73bf1035e1766a65a36640#r76504709>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ARNMdWKbNsx9S9ms-Gbv5wCdeT-GDduBks5qj38GgaJpZM4JuAZl>.